transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java

```diff
@@ -877,7 +848,50 @@ public void testAddReplaceHandlerCalledOnceRegistered() throws Throwable {
         assertTrue(handler2.addedHandler.get());
         assertFalse(handler2.removedHandler.get());
         pipeline.remove(handler2);
-        assertTrue(handler2.removedHandler.get());
+        while (!handler2.removedHandler.get()) {
+            Thread.sleep(10);
+        }
+    }
+
+    @Test(timeout = 3000)
+    public void testAddBefore() throws Throwable {
+        ChannelPipeline pipeline1 = new LocalChannel().pipeline();
+        ChannelPipeline pipeline2 = new LocalChannel().pipeline();
+
+        EventLoopGroup defaultGroup = new DefaultEventLoopGroup(2);
+        try {
+            EventLoop eventLoop1 = defaultGroup.next();
+            EventLoop eventLoop2 = defaultGroup.next();
+
+            eventLoop1.register(pipeline1.channel()).syncUninterruptibly();
+            eventLoop2.register(pipeline2.channel()).syncUninterruptibly();
+
+            for (int i = 0; i < 10; i++) {
+                CountDownLatch latch = new CountDownLatch(2);
```

Comment:
would it be more interesting to have a countdown latch initialized to `10 * 2` declared outside the loop ... submit all the TestTasks ... and await outside the loop ... and assert the items get added to the pipeline in an expected order?